### ISSUE TITLE
Not working with special characters like `===`

### ISSUE THREAD

**Issue by schoero** [rmagatti/alternate-toggler#10]
Hi

I use alternate-toggler with NVIM v0.8.0.
I have added a rule to toggle `===` to `!==` the way you have commented [here](https://github.com/rmagatti/alternate-toggler/issues/6#issuecomment-1235711003).

This does work for normal characters like in `yes` and `no` but not in "special" characters like `===` or `!==`.

The reason for that is that [`<cword>`](https://github.com/rmagatti/alternate-toggler/blob/5575312b949ca5991afa3807175919a0fb448afe/plugin/alternate-toggler.vim#L9) by default does not capture these characters, instead it seems to capture the next valid characters.

```txt
							*word*
A word consists of a sequence of letters, digits and underscores, or a
sequence of other non-blank characters, separated with white space (spaces,
tabs, <EOL>).  This can be changed with the 'iskeyword' option.  An empty line
is also considered to be a word.
							*WORD*
A WORD consists of a sequence of non-blank characters, separated with white
space.  An empty line is also considered to be a WORD.
```

![toggle](https://user-images.githubusercontent.com/42278642/199563879-217046ab-e6da-432f-a7b0-d1905f9ef5ba.gif)

**Comment by schoero** [comment]
I noticed that [changing `<cword>` to `<cWORD>`](https://github.com/rmagatti/alternate-toggler/pull/11#issue-1433569647) has other drawbacks. For example toggling `true` inside `if(something === true)` will capture `true)` and will therefore also not work.

Edit: I found a solution by modifying `iskeyword`.

```lua
vim.opt.iskeyword:append("=")
vim.opt.iskeyword:append("!")
```

Maybe there is a better solution without modifying `iskeyword`, because I fear this has other side effects. 

Otherwise this issue can be closed.

**Comment by rmagatti** [comment]
Hey, thanks for looking into this! I'm not sure there would be an easy way to do it without changing `iskeyword`.
Although maybe something that would mitigate issues that modifying `iskeyword` might have would be:
1. When activating ToggleAlternate, copy the contents of `iskeyword`
2. Change the contents so it considers the right things for the upcoming cword
3. After the command executes, change `iskeyword` back to the original contents.

**Comment by schoero** [comment]
Thank you for your response! You pointed me at the right direction.

But I found a more suitable solution on [stackoverflow](https://stackoverflow.com/a/23315227/10209765). I created a [PR](https://github.com/rmagatti/alternate-toggler/pull/12) that implements this.

It now passed the same text content as you get with `yiw` to the [`LuaAlternateToggle`](https://github.com/rmagatti/alternate-toggler/blob/5575312b949ca5991afa3807175919a0fb448afe/lua/alternate-toggler.lua#L33) function. 

This changes the behavior to what it was originally intended without the sideffects you get by messing with `iskeyword`.

**Comment by rmagatti** [comment]
@schoero I've put some code in a PR inspired by yours that I think should achieve the desired result and I've also made some improvements. Maybe test the branch out before I merge it? I'll likely merge it in in the next 24 hours.

Thanks a ton for putting in the work to figure this out!

**Comment by schoero** [comment]
I installed your branch [`fix-special-characters`](https://github.com/rmagatti/alternate-toggler/tree/fix-special-characters) and tested it quickly. It works in all the cases I tested. I also used the new setup function for configuration, which worked as well.

I also like that it doesn`t change the cursor position anymore 👍.

Another nice thing is that it now works inside quotes, which  I think was [not possible before](https://github.com/rmagatti/alternate-toggler/issues/2#issuecomment-812733714).

I let you know if I notice anything, otherwise it seems good to go.
